src/integrationtests/java/com/aws/greengrass/integrationtests/certificateauthority/CustomCaConfigurationTest.java

```diff
@@ -0,0 +1,201 @@
+
+        return new X509Certificate[]{intermediateCA, rootCA};
+    }
+
+
+    /**
+     * Simulates the request client components make to create their certificates. Clients pass their certificate
+     * chain pem to get verified.
+     */
+    private X509Certificate[] arrangeClientComponentCertificateChain() throws CertificateGenerationException,
+            ExecutionException, InterruptedException, TimeoutException {
+        // Generate Client Certificates
+        AtomicReference<CertificateUpdateEvent> eventRef = new AtomicReference<>();
+
+        Pair<CompletableFuture<Void>, Consumer<CertificateUpdateEvent>> asyncCall =
+                TestUtils.asyncAssertOnConsumer(eventRef::set, 1);
+        GetCertificateRequestOptions requestOptions = new GetCertificateRequestOptions();
+        requestOptions.setCertificateType(GetCertificateRequestOptions.CertificateType.CLIENT);
+        GetCertificateRequest certificateRequest =
+                new GetCertificateRequest("com.aws.clients.Plugin", requestOptions, asyncCall.getRight());
+
+        certificateManager.subscribeToCertificateUpdates(certificateRequest);
+        asyncCall.getLeft().get(1, TimeUnit.SECONDS);
+
+        // Arrange the certificate chain leaf comes first, root comes last
+        return ArrayUtils.addAll(new X509Certificate[]{eventRef.get().getCertificate()},
+                eventRef.get().getCaCertificates());
+    }
+
+
+    @Test
+    void Given_CustomCAConfiguration_WHEN_issuingAClientCertificate_THEN_itsSignedByCustomCA() throws
+            CertificateChainLoadingException, KeyLoadingException, CertificateException, NoSuchAlgorithmException,
+            URISyntaxException, ServiceUnavailableException, OperatorCreationException, CertIOException,
+            InvalidConfigurationException, CertificateGenerationException, ExecutionException, InterruptedException,
+            TimeoutException {
+        X509Certificate[] caCertificates = arrangeCustomCertificateAuthority();
+        X509Certificate[] clientCertificateChain = arrangeClientComponentCertificateChain();
+
+        assertTrue(CertificateTestHelpers.wasCertificateIssuedBy(caCertificates[0], clientCertificateChain[0]));
+    }
+
+    @Test
+    void GIVEN_CustomCAConfiguration_WHEN_whenGeneratingClientCerts_THEN_GGComponentIsVerified() throws
+            NoSuchAlgorithmException, CertificateException, OperatorCreationException, IOException,
+            URISyntaxException, InvalidConfigurationException, KeyLoadingException, ServiceUnavailableException,
+            CertificateChainLoadingException, CertificateGenerationException, ExecutionException, InterruptedException,
+            TimeoutException {
+        this.arrangeCustomCertificateAuthority();
+        X509Certificate[] clientCertificateChain = arrangeClientComponentCertificateChain();
+
+        DeviceAuthClient deviceAuth = new DeviceAuthClient(sessionManagerMock, groupManagerMock, certificateStore);
```

Comment:
Good test. I think `DeviceAuthClient` will eventually go away though. Maybe we can test this through the API interface instead? Or maybe even through IPC